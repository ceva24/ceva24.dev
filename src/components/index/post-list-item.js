import React from "react";
import { css } from "@emotion/react";
import { rhythm } from "src/utils/typography";
import { Link } from "gatsby";
import PostDate from "src/components/post-date";

const PostListItem = ({ node }) => (
    <li
        key={node.id}
        css={css`
            margin-bottom: ${rhythm(1)};
        `}
    >
        <div
            css={css`
                display: flex;
            `}
        >
            <h3
                css={css`
                    margin-bottom: ${rhythm(1 / 3)};
                `}
            >
                <Link
                    to={`/${node.fields.path}`}
                    css={css`
                        text-decoration: none;
                    `}
                >
                    {node.frontmatter.title}
                </Link>
            </h3>
            <div
                css={css`
                    flex-grow: 2;
                    text-align: right;
                    padding-left: ${rhythm(3 / 4)};
                `}
            >
                <PostDate>{node.frontmatter.date}</PostDate>
            </div>
        </div>
        <p>{node.excerpt}</p>
    </li>
);

export default PostListItem;
