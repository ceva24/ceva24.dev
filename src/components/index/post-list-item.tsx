import { css } from "@emotion/react";
import { Link } from "gatsby";
import { rhythm } from "../../utils/typography";
import { PostDate } from "../post-date";

interface Node { 
    node: {
        id: string,
        fields: {
            path: string
        },
        frontmatter: {
            title: string,
            date: string
        },
        excerpt: string
    }
}

const PostListItem: React.FC<Node> = ({ node }: Node) => (
    <li
        key={node.id}
        css={css`
            margin-bottom: ${rhythm(1)};
        `}
    >
        <div
            css={css`
                display: flex;
            `}
        >
            <h3
                css={css`
                    margin-bottom: ${rhythm(1 / 3)};
                `}
            >
                <Link
                    to={`/${node.fields.path}`}
                    css={css`
                        text-decoration: none;
                    `}
                >
                    {node.frontmatter.title}
                </Link>
            </h3>
            <div
                css={css`
                    flex-grow: 2;
                    text-align: right;
                    padding-left: ${rhythm(3 / 4)};
                `}
            >
                <PostDate>{node.frontmatter.date}</PostDate>
            </div>
        </div>
        <p>{node.excerpt}</p>
    </li>
);

export default PostListItem;
