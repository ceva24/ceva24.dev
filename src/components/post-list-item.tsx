import React from "react";
import { Link } from "gatsby";

const PostListItem: React.FC<IndexPageEdge> = ({ node }: IndexPageEdge) => {
    const title = node.frontmatter.title;

    return (
        <li key={node.id} className="mb-0" aria-label={title}>
            <Link
                to={`/${node.fields.path}`}
                aria-label={title}
                className="text-black"
            >
                <div className="p-6 pb-3 hover:bg-gray-50 hover:shadow-lg">
                    <div className="flex">
                        <h3 className="flex-grow text-xl mr-6 uppercase font-normal">
                            {title}
                        </h3>

                        <div className="flex-none">{node.frontmatter.date}</div>
                    </div>

                    <p>{node.excerpt}</p>
                </div>
            </Link>
        </li>
    );
};

export { PostListItem };
