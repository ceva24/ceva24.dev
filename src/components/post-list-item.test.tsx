import { render } from "@testing-library/react";
import React from "react";
import { PostListItem } from "./post-list-item";

describe("post list item", () => {
    it("renders", () => {
        const indexPageEdge = {
            node: {
                id: "1",
                frontmatter: {
                    title: "post title",
                    date: "01-01-2020",
                },
                fields: {
                    slug: "post-title",
                    path: "posts/post-title",
                },
                excerpt: "This is a post",
            },
        };

        const { asFragment } = render(
            <PostListItem node={indexPageEdge.node} />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
