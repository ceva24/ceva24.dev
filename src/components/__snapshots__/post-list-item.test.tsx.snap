// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post list item renders 1`] = `
<DocumentFragment>
  <li>
    <div>
      <h3>
        <a
          href="/posts/post-title"
        >
          post title
        </a>
      </h3>
      <div>
        <strong>
          01-01-2020
        </strong>
      </div>
    </div>
    <p>
      This is a post
    </p>
  </li>
</DocumentFragment>
`;
